5_databases/week1/homework_week1.sql

```diff
@@ -0,0 +1,52 @@
+-- Active: 1706632082876@@localhost@3306@hyf_database
+-- Find out how many tasks are in the task table
+SELECT COUNT(*) AS task_count FROM task;
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(*) AS tasks_without_due_date
+FROM task
+WHERE due_date IS NULL OR due_date < NOW();
+
+-- Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE status_id = (SELECT id FROM status WHERE name = 'Done');
+
+-- Find all the tasks that are not marked as done
+SELECT *
+FROM task
+WHERE status_id <> (SELECT id FROM status WHERE name = 'Done');
+
+-- Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM task
+ORDER BY created DESC;
+
+-- Get the single most recently created task
+SELECT *
+FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+-- Get the title and due date of all tasks where the title or description contains "database"
+SELECT title, due_date
+FROM task
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+-- Get the title and status (as text) of all tasks
+SELECT t.title, s.name AS status
```

Comment:
This syntax is valid (t.name, s.id, etc.) however I would not suggest using it a lot. Especially when you'll have a big database with a lot of tables. Your table names are pretty much short and straightforward but explanatory. And 1 letter is not😊